homeassistant/components/tplink/light.py

```diff
@@ -213,13 +219,31 @@ def update(self):
                 except KeyError:
                     # device returned no daily/monthly history
                     pass
+        except (SmartDeviceException, OSError) as ex:
+            _LOGGER.warning(
+                "Retrying in %s for %s|%s due to: %s",
+                SLEEP_TIME,
+                self.smartbulb.host,
+                self._alias,
+                ex,
+            )
+        self._is_ready = True
```

Comment:
Now we'll always be ready no matter if there's an exception or not. Shouldn't we take different actions depending on exception happening?

Reply:
I don't have lights to confirm so I didn't change the exception handling for lights, but for the switches I only have seen the connection refused exception. From the comments on other threads it seemed to me that the lights had the same issue...just connection refused. It would make sense to me that the lights could have other issues since they have more functionality like changing color, but not sure how to validate that.

Reply:
We're setting `self._is_ready` to `True` in the normal flow here, but we're not stopping this normal flow in the `except` block, so `self._is_ready` will be `True` also after an exception. I'm thinking we should make sure it's `False` after an exception.

Reply:
That was intentional because the is_ready is just for the for loop...so if there is an exception it should just continue in the for loop which will reset it to false.

Reply:
That was intentional because the is_ready is just for the for loop...so if there is an exception it should just continue in the for loop which will reset it to false.

Reply:
But the loop won't continue since we're catching the exception here and setting `is_ready` to `True`.

Reply:
We need to add a `return` in the except block after the warning.

Reply:
Can't believe I missed that, I moved it to the end of the inside the try and added a return at the end of the function